libs/island-ui/core/src/lib/ModalBase/ModalBase.tsx

```diff
@@ -71,6 +76,10 @@ export const ModalBase: FC<ModalBaseProps> = ({
     if (toggleClose) closeModal()
   }, [toggleClose])
 
+  useEffect(() => {
+    if (!modal.visible && onClose) onClose()
```

Comment:
Would it make sense to have this something like?
```suggestion
    onChange(modal.visible)
```
or maybe `onVisibilityChange`, just so we won't need to extend this in the future and change the API, or will that not work for your use case?

Reply:
Good point 👍